src/Microsoft.ML.Data/DataLoadSave/Database/DatabaseSource.cs

```diff
@@ -25,6 +25,26 @@ public DatabaseSource(DbProviderFactory providerFactory, string connectionString
             CommandText = commandText;
         }
 
+        /// <summary>Creates a new instance of the <see cref="DatabaseSource" /> class.</summary>
+        /// <param name="providerFactory">The factory used to create the <see cref="DbConnection"/>..</param>
+        /// <param name="connectionString">The string used to open the connection.</param>
+        /// <param name="commandText">The text command to run against the data source.</param>
+        /// <param name="commandTimeOut">The timeout(in seconds) for database command.</param>
+        public DatabaseSource(DbProviderFactory providerFactory, string connectionString, string commandText, int commandTimeOut)
+        {
```

Comment:
Ideally, the previous constructor would call this constructor with a default timeout value and the two constructors would have a shared implementation.

Reply:
As I mentioned above, I think it would be better if the constructor without the constructor called the constructor with the timeout (with a constant value). That way the CommandTimeOutInSeconds would always be initialized and you wouldn't need a nullable type and you wouldn't need a separate HasValue check in DatabaseSource.cs.

---
In reply to: [451066115](https://github.com/dotnet/machinelearning/pull/5288#discussion_r451066115) [](ancestors = 451066115)